pubsub/api/src/create_topic_with_cloud_storage_ingestion.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * Copyright 2025 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * For instructions on how to run the full sample:
+ *
+ * @see https://github.com/GoogleCloudPlatform/php-docs-samples/blob/main/pubsub/api/README.md
+ */
+
+namespace Google\Cloud\Samples\PubSub;
+
+# [START pubsub_create_topic_with_cloud_storage_ingestion]
+use Google\Cloud\PubSub\PubSubClient;
+use Google\Cloud\PubSub\V1\IngestionDataSourceSettings\CloudStorage\AvroFormat;
+use Google\Cloud\PubSub\V1\IngestionDataSourceSettings\CloudStorage\PubSubAvroFormat;
+use Google\Cloud\PubSub\V1\IngestionDataSourceSettings\CloudStorage\TextFormat;
+use Google\Protobuf\Timestamp;
+
+/**
+ * Creates a topic with Cloud Storage Ingestion.
+ *
+ * @param string $projectId  The Google project ID.
+ * @param string $topicName  The Pub/Sub topic name.
+ * @param string $bucket  Cloud Storage bucket.
+ * @param string $inputFormat  Input format for the Cloud Storage data. Must be one of text, avro, or pubsub_avro.
+ * @param string $textDelimeter  Delimiter for text format input.
+ * @param string $matchGlob  Glob pattern used to match objects that will be ingested. If unset, all objects will be ingested.
+ * @param string $minimumObjectCreatedTime  Only objects with a larger or equal creation timestamp will be ingested.
+ */
+function create_topic_with_cloud_storage_ingestion(
+    string $projectId,
+    string $topicName,
+    string $bucket,
+    string $inputFormat,
+    string $textDelimeter,
+    string $matchGlob,
+    string $minimumObjectCreatedTime
+): void {
+    $datetime = new \DateTimeImmutable($minimumObjectCreatedTime);
+    $timestamp = (new Timestamp())->setSeconds($datetime->getTimestamp())->setNanos($datetime->format('u') * 1000);
+
+    $cloudStorageData = [
+        'bucket' => $bucket,
+        'minimum_object_create_time' => $timestamp
+    ];
+
+    switch ($inputFormat) {
+        case 'text':
+            $cloudStorageData['text_format'] = new TextFormat(['delimiter' => $textDelimeter]);
+            break;
+        case 'avro':
+            $cloudStorageData['avro_format'] = new AvroFormat();
+            break;
+        case 'pubsub_avro':
+            $cloudStorageData['pubsub_avro_format'] = new PubSubAvroFormat();
+            break;
+        default:
+            throw new \InvalidArgumentException('inputFormat must be in (\'text\', \'avro\', \'pubsub_avro\'); got value: ' . $inputFormat);
+    }
```

Comment:
nit: maybe would look cleaner using `match` instead of `switch`, but this is subjective so it's up to you!
```suggestion
    $cloudStorageData[$inputFormat . '_format'] = match($inputFormat) {
        'text' => new TextFormat(['delimiter' => $textDelimiter]),
        'avro' => new AvroFormat(),
        'pubsub_avro' => new PubSubAvroFormat(),
        default => throw new \InvalidArgumentException(
            'inputFormat must be in (\'text\', \'avro\', \'pubsub_avro\'); got value: ' . $inputFormat
        )
    };
```